features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java

```diff
@@ -261,12 +261,22 @@ public LocationAwarePollerClient getLocationAwarePollerClient() {
     protected void reportResult(final String locationName, final RemotePolledService polledService, final PollStatus pollResult) {
         final OnmsMonitoringLocation location = this.monitoringLocationDao.get(locationName);
 
-        final OnmsLocationSpecificStatus status = new OnmsLocationSpecificStatus();
-        status.setLocation(location);
-        status.setMonitoredService(polledService.getMonSvc());
-        status.setPollResult(pollResult);
+        final OnmsLocationSpecificStatus oldLocationSpecificStatus = this.locationSpecificStatusDao.getMostRecentStatusChange(location, polledService.getMonSvc());
```

Comment:
The whole select/update cycle should run in a single transaction

Reply:
done.